### PR TITLE
update to latest elixir and erlang

### PR DESCRIPTION
**Pull request by happysalada**
### Changes

This updates the tool-versions
Elixir to 1.12.1
OTP to 24.0.2
node to 16.3.0
I've checked compilation on local.
I've been using those tools with another project in production and didn't observer major problems.

Note that I didn't update the docker image since the tags from the origin image are old
https://hub.docker.com/r/hexpm/elixir/tags?page=1&ordering=last_updated&name=alpine
Elixir 1.12.1 isn't there
alpine is old too, latest is at 3.13.5

### Tests
- [ ] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [ ] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [ ] This change does not need a documentation update

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,3 @@
-elixir 1.11.3-otp-23
-erlang 23.2.1
-nodejs 15.3.0
+v1.12.1-otp-24
```
Comment (ukutaht) [inline .tool-versions:1]: this doesn't seem right.. maybe that's why the build is failing? Should be:

```
elixir 1.12.1-otp-24
```

Comment (happysalada) [reply .tool-versions:1]: Oh, my bad

Comment (happysalada) [reply .tool-versions]: should be good to go

**Comment by happysalada** [comment]
Just checked and the failure is mix format on non related files

**Comment by ukutaht** [comment]
Is there any rush with this? I'd prefer to upgrade when the new Docker image is available as well

**Comment by happysalada** [comment]
No rush at all.

**Comment by ukutaht** [comment]
Closing in favour of #1169

**Pull request closed without merging**